### PR TITLE
Cleanup and pull in recent Java SDK features

### PR DESCRIPTION
**Pull request by AnotherDaniel**
- clean up and update main readme
- remove contributors list (not needed, have git logs)
- pull in changes and additions from Java SDK PR #60

### CODE REVIEW

File: src/proto/cloudevents/protocloudevent.rs
Comment (PLeVasseur) [inline src/proto/cloudevents/protocloudevent.rs:27]: IMHO it'd be nice if we're only using `LongUriSerializer` to only have to use that one, i.e. only have to:
```rust
use crate::uri::serializer::{LongUriSerializer};
```

I think we can achieve that by [modifying this line](https://github.com/eclipse-uprotocol/up-rust/blob/ae311dd9d170d7030344fb020a9ed83342f6b0d6/src/uri/serializer/longuriserializer.rs#L19):
```rust
use crate::uri::serializer::{SerializationError, UriSerializer};
```
to
```rust
pub use crate::uri::serializer::{SerializationError, UriSerializer};
```

File: src/proto/cloudevents/protocloudevent.rs
Comment (PLeVasseur) [inline src/proto/cloudevents/protocloudevent.rs:31]: I noticed leaning on getting default and unwraps throughout.

I'm wondering if we should be doing any additional error checking and make it TryFrom instead to reflect that it may fail?

WDYT?

File: src/rpc/rpcclient.rs
Comment (PLeVasseur) [inline src/rpc/rpcclient.rs:49]: :+1:

File: src/rpc/rpcmapper.rs
Comment (PLeVasseur) [inline src/rpc/rpcmapper.rs:126]: Funnily enough, I was doing the same thing [over in this PR](https://github.com/eclipse-uprotocol/up-rust/pull/27) :laughing: 

We should probably be a little more careful here, [similar to what I did on that PR](https://github.com/eclipse-uprotocol/up-rust/pull/27/files#diff-8d66726b55938f3831ad9ba37cb66af6c321e6091a883099cde61b586056cbc3R86) s.t. we don't accidentally panic.

File: src/rpc/rpcmapper.rs
Comment (PLeVasseur) [inline src/rpc/rpcmapper.rs:307]: Again, funny we both did this haha

Kai gave some good feedback on that PR on how to handle this nicely, perhaps you could [reference that](https://github.com/eclipse-uprotocol/up-rust/pull/27/files#diff-8d66726b55938f3831ad9ba37cb66af6c321e6091a883099cde61b586056cbc3R130-R136)?

File: src/uri/validator/urivalidator.rs
Comment (PLeVasseur) [inline src/uri/validator/urivalidator.rs:234]: This one I would pause on tho. IMHO it makes more sense to merge the [PR I've got going](https://github.com/eclipse-uprotocol/up-rust/pull/18) for this functionality. That PR goes into a lot more detail in checking also that the fields from the UUri are able to fit the specified number of bits & bytes available.

If the PR I have for checking micro form goes through, that can then be mirrored back to `up-java`.

**Comment by PLeVasseur** [comment]
Overall good to stay in sync with `up-java`, but I'd recommend removing [this bit here for `is_micro_form_authority()`](https://github.com/eclipse-uprotocol/up-rust/pull/28#discussion_r1471866303) since [this other PR](https://github.com/eclipse-uprotocol/up-rust/pull/18) I have does further checking into buffer size & available number of bytes.

If you could remove that bit, I think we're good :+1:

**Review by stevenhartley — CHANGES_REQUESTED** [review]
Mapping error for priority that needs fixing

File: src/proto/cloudevents/protocloudevent.rs
Comment (AnotherDaniel) [reply src/proto/cloudevents/protocloudevent.rs:27]: Good catch!

File: src/rpc/rpcmapper.rs
Comment (AnotherDaniel) [reply src/rpc/rpcmapper.rs:126]: Pulled that in - and apologies for waltzing over your work there!

File: src/uri/validator/urivalidator.rs
Comment (AnotherDaniel) [reply src/uri/validator/urivalidator.rs]: Removed!

File: src/proto/cloudevents/protocloudevent.rs
Comment (AnotherDaniel) [reply src/proto/cloudevents/protocloudevent.rs:27]: Addendum - this doesn't really work for some reason, dependent code still won't compile without explicitly use-ing the Trait implementation. I guess I'll revert this change for now.

File: src/proto/cloudevents/protocloudevent.rs
Comment (AnotherDaniel) [reply src/proto/cloudevents/protocloudevent.rs:57]: I made the changes to achieve this - not yet based off the referenced PR, but manually. Will change that bit as soon as up-core-api#98 is merged.

File: src/proto/cloudevents/protocloudevent.rs
Comment (PLeVasseur) [reply src/proto/cloudevents/protocloudevent.rs:27]: My bad! Looks like this is not possible in Rust. We have to bring the trait into scope explicitly with a `use`. Seems this is a conscious design decision to help when multiple traits define things with the same names and helps readability to understand where a function could be coming from.

Comment (AnotherDaniel) [reply src/proto/cloudevents/protocloudevent.rs:27]: heh - forget all that, found the problem. We will have a try_from.

File: src/proto/cloudevents/protocloudevent.rs
Comment (AnotherDaniel) [reply src/proto/cloudevents/protocloudevent.rs:34]: we have a try_from now - However, next I will have to spend some quality time revisiting the test setup for cloudevent.rs, stay tuned for another change in that area.

**Comment by AnotherDaniel** [comment]
Will take a couple of days yet - need to revisit some test cases at the very least, plus want to wait for up-core-api 1.5.6
Will let you know once this is ready!

**Comment by stevenhartley** [comment]
Yes I will do a release this week, not waiting for the other changes that I want to review at the generic uProtocol meeting on Thursday.

File: src/transport/datamodel/utransport.rs
Comment (AnotherDaniel) [reply src/transport/datamodel/utransport.rs:54]: Agree and fixed

File: src/transport/datamodel/utransport.rs
Comment (sophokles73) [inline src/transport/datamodel/utransport.rs:50]: The doc comment still refers to the original types. Which also is a good example for why you should not mention the types of parameters in the doc comments at all (they are obvious from the generated documentation anyway) ;-)

**Comment by sophokles73** [comment]
@AnotherDaniel There is no doubt that the changes in this PR are relevant and helpful :+1:
However, the PR contains quite a few changes which do not necessarily seem to be related to each other. This makes it quite hard to review and rebase, because it touches on so many areas. For the future, I would suggest to try to keep the scope of a PR small and consistent and try to only change things that are in scope. That will make it much easier to get things reviewed and merged.

**Comment by AnotherDaniel** [comment]
I'm still mentally working my way down from the all-encompassing initial commit, but point taken :)

**Comment by AnotherDaniel** [comment]
@PLeVasseur , I had to do a number on the MicroUriSerializer code, to make it work with up-core-api v1.5.6  -   sorry to get in the way of of your #18 there, probably you can update that one to the same up-core-api and then take over my changes?
@sophokles73 I'm on the fence about the remaining UAuthority::get_name()

**Comment by sophokles73** [comment]
Why do you want to remove the getters from UAuthority in the first place? Aren't they useful anymore?

**Comment by AnotherDaniel** [comment]
They now come with the generated code...

**Comment by sophokles73** [comment]
@AnotherDaniel can you rebase this PR?

**Comment by PLeVasseur** [comment]
Hey folks -- sorry I missed that the up-core-api change got merged fixing the ambiguity around id/ip. I'll have a go at updating this.

cc @sophokles73 @AnotherDaniel

**Review by sophokles73 — APPROVED** [review]
LGTM

**Comment by AnotherDaniel** [comment]
@stevenhartley the changes you requested wrt up-core-api 1.5.6 (umessage option strings) are in now

**Pull request merged**
